docs/flow/reference/config-blocks/_index.md

```diff
@@ -6,3 +6,13 @@ weight: 200
 ---
 
 # Configuration blocks
+
+Configuration blocks are optional top-level blocks which can be used to
```

Comment:
Should we add why that cant use other components?

Reply:
Isn't the answer implied below?

> Configuration blocks are _not_ components, so expressions which reference
 components are invalid. Expressions which do not reference components (e.g., `env("LOG_LEVEL")`) are permitted.
 
This implies that only components can reference other components, but I can make that more explicit if you'd like

Reply:
I was thinking of more of the "why" they aren't components. IE that these are evaluated before/differently than the component system.

Reply:
The use case I am thinking of is: the user asks why can't I change the log level dynamically. It might be good to state the underlying reason to avoid issues being opened to make config blocks components. 

```
logging {
 level = local.file.content
}
local.file "log" {
  path = "/tmp/deubg.level" 
}
```

Reply:
Hmm, I'm not sure, that feels like an implementation detail which will probably change. I'd like to suggest that we leave an explanation out.

You've also mentioned in the past wanting to make all config blocks components, so this is probably something that won't last very long.

Reply:
I do think this separation won't last forever. My thought is because this behaves so differently from the rest of the flow subsystem that, it should have an explicit call out. If you don't think it's worth it, I am okay with not having the details at this time. If we get questions or it's unclear, we can revisit later.

Reply:
Yeah, I'd personally like to avoid mentioning any implementation details in our user documentation. +1 on revisiting later if it starts coming up